rotkehlchen/chain/ethereum/modules/liquity/trove.py

```diff
@@ -0,0 +1,506 @@
+                    spent_balance = Balance(
+                        amount=abs(collateral_change),
+                        usd_value=query_usd_price_or_use_default(
+                            A_ETH,
+                            timestamp,
+                            ZERO,
+                            'Liquity',
+                        ),
+                    )
+                    pnl = [AssetBalance(asset=A_ETH, balance=-spent_balance)]
+                event = DefiEvent(
+                    timestamp=Timestamp(change['transaction']['timestamp']),
+                    wrapped_event=change,
+                    event_type=DefiEventType.LIQUITY,
+                    got_asset=got_asset,
+                    got_balance=got_balance,
+                    spent_asset=spent_asset,
+                    spent_balance=spent_balance,
+                    pnl=pnl,
+                    count_spent_got_cost_basis=count_spent_got_cost_basis,
+                    tx_hash=change['transaction']['id'],
+                )
+                events.append(event)
+            except KeyError as e:
+                self.msg_aggregator.add_warning(
+                    f'Ignoring Liquity Trove event in Liquity. '
+                    f'Failed to decode remote information. {str(e)}.',
+                )
+                continue
+        return events
+
+    def _get_raw_history(
+        self,
+        addresses: List[ChecksumEthAddress],
+        query_for: Literal['stake', 'trove'],
+    ) -> Dict[str, Any]:
+
+        param_types = {
+            '$addresses': '[Bytes!]',
+        }
+        param_values = {
+            'addresses': [addr.lower() for addr in addresses],
+        }
+        if query_for == 'trove':
+            querystr = format_query_indentation(QUERY_TROVE)
+        else:
+            querystr = format_query_indentation(QUERY_STAKE)
+        return self.graph.query(
+            querystr=querystr,
+            param_types=param_types,
+            param_values=param_values,
+        )
+
+    def get_history(
+        self,
+        addresses: List[ChecksumEthAddress],
+        from_timestamp: Timestamp,
+        to_timestamp: Timestamp,
+    ) -> Dict[ChecksumEthAddress, Dict[str, List[LiquityEvent]]]:
+        query = self._get_raw_history(addresses, 'trove')
+        staked = self._get_raw_history(addresses, 'stake')
+        result: Dict[ChecksumEthAddress, Dict[str, List[LiquityEvent]]] = defaultdict(lambda: defaultdict(list))  # noqa: E501
+        for trove in query['troves']:
+            owner = to_checksum_address(trove['owner']['id'])
+            for change in trove['changes']:
+                try:
+                    timestamp = change['transaction']['timestamp']
+                    if timestamp < from_timestamp:
+                        continue
+                    if timestamp > to_timestamp:
+                        break
+                    operation = TroveOperation.deserialize(change['troveOperation'])
```

Comment:
What happens if this and all other `deserialize` functions raise? I see you only catch `KeyError`